lib/api/movies.ts

```diff
@@ -0,0 +1,117 @@
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
```

Comment:
let's use the axios instance and set the header configs and interceptors to handle the auth token logic